### PR TITLE
Premium; settings screen and basic product scaffold

### PR DESCRIPTION
**Pull request by jedireza**
This PR adds the ability to purchase Neeva Premium from inside app settings.

### Screenshots

Here is the end-to-end flow on my local dev cluster.

https://user-images.githubusercontent.com/979929/179628311-2c870bdd-e38f-4285-8902-8cac3968cf54.mp4

If a user is a premium subscriber, but doesn't pay through Apple, we let them know to login to desktop to manage their subscription.

### CODE REVIEW

File: Client/Frontend/Settings/Neeva/NeevaPremium.swift [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+                    .SettingPremiumNoProductsFound,
+                    attributes: EnvironmentHelper.shared.getAttributes()
+                )
+            }
+        }
+    }
+
+    private var productList: some View {
+        List(products) { product in
+            Button(action: { purchase(product, reloadUserInfo: !isSubScribedToPlan(product.id)) }) {
```
Comment (chung-lai) [inline Client/Frontend/Settings/Neeva/NeevaPremium.swift:69]: `isSubScribedToPlan` -> `isSubscribedToPlan`

File: Client/Frontend/Settings/Neeva/NeevaAccountInfoView.swift [hunk 2]
```diff
@@ -37,78 +38,31 @@ struct NeevaAccountInfoView: View {
...
-                    Button("Learn More on the Neeva Website") {
-                        openURL(NeevaConstants.appMembershipURL)
+                // Only show for iOS 15 users who are not Lifetime and have not paid through another source
+                if #available(iOS 15, *) {
+                    if userInfo.subscriptionType != .lifetime
+                        && (userInfo.subscription?.source == SubscriptionSource.none
+                            || userInfo.subscription?.source == SubscriptionSource.apple)
+                    {
+                        NavigationLink(
+                            destination: NeevaPremium(userInfo: userInfo),
```
Comment (chung-lai) [inline Client/Frontend/Settings/Neeva/NeevaAccountInfoView.swift:51]: Rename `NeevaPremium` -> `NeevaPremiumView`?

**Review by chung-lai — APPROVED** [review]
Went through the PR offline and addressed the feedback about:
- iOS 14 support
- re-subscription
- some refactorings for later
- potential testflight issues after pushing
- some logging attribute related changes

The PR itself looks good and should move to do more testflight testing.

File: Client/Frontend/Settings/Neeva/NeevaPremium.swift (hunk 1 above)
Comment (jedireza) [reply Client/Frontend/Settings/Neeva/NeevaPremium.swift]: Good catch. Fixed.

File: Client/Frontend/Settings/Neeva/NeevaAccountInfoView.swift (hunk 2 above)
Comment (jedireza) [reply Client/Frontend/Settings/Neeva/NeevaAccountInfoView.swift]: Good catch. Fixed.

**Pull request merged**